Sources/SQLite/Typed/Query.swift

```diff
@@ -494,8 +494,9 @@ extension QueryType {
             return nil
         }
 
-        return " ".join(clauses.join.map { type, query, condition in
-            " ".join([
+        return " ".join(clauses.join.map { arg in
+            let (type, query, condition) = arg
+            return " ".join([
```

Comment:
That's the issue in #664 : the map closure take one tuple argument instead of its components. That's new in Swift 4, see [SE-0110](https://github.com/apple/swift-evolution/blob/master/proposals/0110-distingish-single-tuple-arg.md).